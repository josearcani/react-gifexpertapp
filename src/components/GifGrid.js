import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';

import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs( category );

  return (
    <>
      <h3 className="card-grid animate__animated animate__fadeInDown">{ category }</h3>
      { loading && <p className="animate__animated animate__pulse animate__infinite">Cargando...</p> }
      <div className="card-grid animate__animated animate__fadeIn">
        { images.map( img  => (
            <GifGridItem
              key={ img.id }
              { ...img } />
          ))
        } 
      </div>
    </>
  )
}

export default GifGrid;