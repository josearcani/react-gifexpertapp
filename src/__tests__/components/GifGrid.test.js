import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs'); // fingir la llamada y simular la data

describe('Pruebas en <GifGrid />', () => {

  const category = 'Goku';

  test('debe mostrar <GifGrid /> correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })
    const wrapper = shallow(<GifGrid category={ category } />)
    expect(wrapper).toMatchSnapshot();
  })

  // simular que ya tengo la información del useFetchGifs
  test('debe de mostrar items cuando se cargan imágenes de useFetchGifs', () => {
    // hacemos un mock
    const gifs = [{
      id: '1234biafsd',
      url: 'https://locahost/anything/cheap.jpg',
      title: 'Love fo my life'
    },
    {
      id: '1123biafsd',
      url: 'https://locahost/anything/cheap.jpg',
      title: 'Love fo my life'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    })
    const wrapper = shallow(<GifGrid category={ category } />);
    // no existe el <p> de loading
    // expect(wrapper).toMatchSnapshot(); // util para ver el contenido generado
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length); // cuantos de esos componentes hay

  })



})