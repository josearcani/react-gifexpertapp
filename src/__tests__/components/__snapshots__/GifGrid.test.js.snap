// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="card-grid animate__animated animate__fadeInDown"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__pulse animate__infinite"
  >
    Cargando...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
